ocaml/xapi/import.ml

```diff
@@ -247,10 +247,7 @@ let assert_can_restore_backup ~__context rpc session_id (x : header) =
       List.iter
         (fun (mac', vm') ->
           if mac = mac' && not (is_compatible vm vm') then
-            raise
-              (Api_errors.Server_error
-                 (Api_errors.duplicate_vm, [vm'.API.vM_uuid])
-              )
+            raise Api_errors.(Server_error (duplicate_mac_seed, [mac']))
```

Comment:
It also tests for other incompatibilities with `is_compatible`, so maybe split the cases and raise separate errors.

Reply:
The check is intertwined, (it's an `&&`, not `||`), there's something that's missing here from the report.

Reply:
`is_compatible` checks whether the 2 vms are the same, they are snapshots of the same vm or one is a snapshot of the other.
This means that the only when there are two different VMs relates we care that the mac is the same, this is why the duplicate mac is the correct error to raise.

Reply:
`is_compatible` checks whether the 2 vms are the same, they are snapshots of the same vm or one is a snapshot of the other.
This means that the only when there are two different VMs relates we care that the mac is the same, this is why the duplicate mac is the correct error to raise.